tensorflow/compiler/xla/service/gpu/cudnn_conv_rewriter_test.cc

```diff
@@ -631,6 +642,46 @@ TEST_F(CudnnConvRewriterTest, BackwardInputConvolveConstantFilter) {
                           0));
 }
 
+// Check that an integer forward convolution is replaced with a custom call to
+// cudnnConvolutionForward
+TEST_F(CudnnConvRewriterTest, ForwardIntegerConvolution) {
+  const string module_str = absl::StrFormat(R"(
+    HloModule Test
+
+    ENTRY Test {
+      input = s8[1,17,3,3] parameter(0)
+      filter = s8[3,3,17,63] parameter(1)
+
+      ROOT conv = f32[1,63,3,3] convolution(input, filter), window={size=3x3 pad=1_1x1_1}, dim_labels=bf01_01io->bf01, feature_group_count=1
+    })");
+  TF_ASSERT_OK_AND_ASSIGN(auto m, ParseAndReturnVerifiedModule(module_str));
+
+  EXPECT_TRUE(RunPass(m.get()));
+  EXPECT_THAT(
+      m->entry_computation()->root_instruction(),
+      op::GetTupleElement(
+          op::CustomCall(kCudnnConvForwardCallTarget, _, op::Parameter()), 0));
+
+  string optimized_hlo_string = GetOptimizedHlo(module_str);
+  EXPECT_TRUE(RunAndCompare(module_str, ErrorSpec{0.01}))
+      << optimized_hlo_string;
+}
+
+// Check that an integer forward convolution is not allowed
+TEST_F(CudnnConvRewriterTest, ForwardIntegerOutputConvolution) {
+  const string module_str = absl::StrFormat(R"(
+    HloModule Test
+
+    ENTRY Test {
+      input = s8[1,2,3,3] parameter(0)
+      filter = s8[3,3,2,5] parameter(1)
+
+      ROOT conv = s8[1,5,3,3] convolution(input, filter), window={size=3x3 pad=1_1x1_1}, dim_labels=bf01_01io->bf01, feature_group_count=1
+    })");
+  TF_ASSERT_OK_AND_ASSIGN(auto m, ParseAndReturnVerifiedModule(module_str));
+
+//   ASSERT_FALSE(CudnnConvRewriter().Run(m.get()).ValueOrDie());
```

Comment:
I want to have a test point to lock down the error for in8-to-int8 convolution.  This is not complete.  I need help to find an example.

Reply:
What I meant to say was just a test.  I was looking for a test example to lock down a designed error message.